diff --git a/setup.py b/setup.py
index 4708b2a81..9e3e06ada 100644
--- a/setup.py
+++ b/setup.py
@@ -87,6 +87,9 @@ def load_config(config_path: Path) -> Dict[str, str | List[Dict[str, str]]]:
     return parse_config_vars(config_data)
 
 
+pyspec = {}
+
+
 def build_spec(
     fork: str,
     preset_name: str,
@@ -113,6 +116,13 @@ def build_spec(
             spec_object.custom_types | spec_object.preset_dep_custom_types,
         )
 
+    if preset_name not in pyspec:
+        pyspec[preset_name] = {}
+    if fork not in pyspec[preset_name]:
+        pyspec[preset_name][fork] = {}
+    for key, value in spec_object._asdict().items():
+        pyspec[preset_name][fork][key] = value
+
     return objects_to_spec(preset_name, spec_object, fork, class_objects)
 
 
@@ -215,6 +225,11 @@ class PySpecCommand(Command):
                 # `mainnet` is the default spec.
                 out.write("from . import mainnet as spec  # noqa:F401\n")
 
+        with open("pyspec.json", "w") as file:
+            import json
+
+            json.dump(pyspec, file)
+
 
 class BuildPyCommand(build_py):
     """Customize the build command to run the spec-builder on setup.py build"""
